symengine/lib/symengine_wrapper.pyx

```diff
@@ -2640,35 +2643,32 @@ cdef class Lambdify(object):
                 e_ = sympify(e)
                 outs_.push_back(e_.thisptr)
 
-        if real:
-            self.lambda_double.resize(1)
-            self.lambda_double[0].init(args_, outs_)
-        else:
-            self.lambda_double_complex.resize(1)
-            self.lambda_double_complex[0].init(args_, outs_)
-
+        self._init(args_, outs_)
 
-    cdef void _eval(self, ValueType[::1] inp, ValueType[::1] out):
-        cdef size_t idx, ninp = inp.size, nout = out.size
+    cdef _init(self, symengine.vec_basic& args_, symengine.vec_basic& outs_):
+        raise ValueError("Not supported")
 
-        if inp.size != self.args_size:
-            raise ValueError("Size of inp incompatible with number of args.")
-        if out.size != self.out_size:
-            raise ValueError("Size of out incompatible with number of exprs.")
+    cdef void _eval_real(self, double[::1] inp, double[::1] out):
+        raise ValueError("Not supported")
 
-        # Convert expr_subs to doubles write to out
-        if ValueType == cython.double:
-            self.lambda_double[0].call(&out[0], &inp[0])
-        else:
-            self.lambda_double_complex[0].call(&out[0], &inp[0])
+    cdef void _eval_complex(self, double complex[::1] inp, double complex[::1] out):
+        raise ValueError("Not supported")
 
     # the two cpdef:ed methods below may use void return type
     # once Cython 0.23 (from 2015) is acceptable as requirement.
     cpdef unsafe_real(self, double[::1] inp, double[::1] out):
```

Comment:
This checking was already there in `_eval` which was called previously by `unsafe_real`. We can rename this to `eval_real` and this would call `_eval_real` (which should changed into a cpdef method)